src/components/tikzpictureview.ts

```diff
@@ -0,0 +1,366 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as fse from 'fs-extra'
+import * as child_process from 'child_process'
+import { tmpdir } from 'os'
+import { promisify } from 'util'
+
+import { Extension } from '../main'
+import { stripComments } from '../utils'
+
+const removeDir = promisify(fse.remove)
+
+interface IFileTikzCollection {
+    location: string
+    tempDir: string
+    tikzPictures: IFileTikzPicture[]
+    preamble: string
+    lastChange: number
+}
+
+interface IFileTikzPicture {
+    range: vscode.Range
+    content: string
+    tempFile: string
+    lastChange: number
+}
+
+export class TikzPictureView {
+    extension: Extension
+    tikzCollections: { [filePath: string]: IFileTikzCollection } = {}
+    initalised = false
+
+    private TEMPFOLDER_NAME = 'vscode-latexworkshop'
+
+    constructor(extension: Extension) {
+        this.extension = extension
+    }
+
+    public async view(document: vscode.TextDocument, range: vscode.Range) {
+        if (!this.initalised) {
+            await this.cleanupTempDir()
+            if (!fs.existsSync(path.join(tmpdir(), this.TEMPFOLDER_NAME))) {
+                await fs.mkdirSync(path.join(tmpdir(), this.TEMPFOLDER_NAME))
+            }
+            this.initalised = true
+        }
+
+        const fileTikzCollection = await this.getFileTikzCollection(document)
+
+        const tikzPicture = await this.getTikzEntry(fileTikzCollection, {
+            range,
+            content: document.getText(range)
+        })
+
+        const preambleStatus = await this.checkPreamble(fileTikzCollection)
+
+        if (preambleStatus === 'stable') {
+            this.updateTikzPicture(tikzPicture)
+        }
+    }
+
+    public async onFileChange(
+        document: vscode.TextDocument,
+        changes: vscode.TextDocumentContentChangeEvent[],
+        waitedDelay?: boolean
+    ) {
+        function processModificationToTikzPicture(
```

Comment:
Define `processModificationToTikzPicture` as the private function of `class TikzPictureView`

Reply:
Done in 534dbfdfb46746c1629176b4ad2264b73f449214. Though TBH I don't see much point in in being a module function in it's own right…